x-pack/platform/plugins/shared/streams/server/lib/streams/client.ts

```diff
@@ -528,10 +537,13 @@ export class StreamsClient {
         REQUIRED_MANAGE_PRIVILEGES.every((privilege) => privileges.cluster[privilege] === true) &&
         Object.values(privileges.index[name]).every((privilege) => privilege === true),
       monitor: privileges.index[name].monitor,
-      lifecycle:
-        privileges.index[name].manage_data_stream_lifecycle && privileges.index[name].manage_ilm,
+      // on serverless, there is no ILM, so we map lifecycle to true if the user has manage_data_stream_lifecycle
+      lifecycle: isServerless
+        ? privileges.index[name].manage_data_stream_lifecycle
+        : privileges.index[name].manage_data_stream_lifecycle && privileges.index[name].manage_ilm,
       simulate: privileges.cluster.read_pipeline && privileges.index[name].create,
-      text_structure: privileges.cluster.monitor_text_structure,
+      // text structure is always available for the internal user, but not for the current user
+      text_structure: isServerless ? true : privileges.cluster.monitor_text_structure,
```

Comment:
This wouldn't still cover from failures if we check the privileges, but also use the current user right?

Reply:
We don't, but the check for privileges gates the API on stateful for users who don't have those privileges.
We can also allow that for stateful deployments, even if the user doesn't have `monitor_text_structure` privileges, in that case this can be removed.

Reply:
Discussed offline, let's leave the check in for stateful at this time, we can get back to it.

> This wouldn't still cover from failures if we check the privileges, but also use the current user right?

You mean on serverless? Yes, that's true, it's slightly confusing... this isn't so much about `text_structure` capabilities but `date_format` capabilities via the streams API (which hinge on `text_structure` capabilities on stateful, but not on serverless). But I don't think we need to change it right now - let's check back with Luca next week and ideally we just remove it altogether.